omod/src/main/java/org/openmrs/module/ugandaemrreports/web/reports/WebRendererReportsController.java

```diff
@@ -0,0 +1,100 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.ugandaemrreports.web.reports;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.htmlwidgets.web.WidgetUtil;
+import org.openmrs.module.reporting.common.ReflectionUtil;
+import org.openmrs.module.reporting.dataset.definition.DataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.service.DataSetDefinitionService;
+import org.openmrs.module.reporting.definition.DefinitionUtil;
+import org.openmrs.module.reporting.definition.configuration.Property;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.report.*;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.definition.service.ReportDefinitionService;
+import org.openmrs.module.reporting.report.renderer.CsvReportRenderer;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+@Controller
+public class WebRendererReportsController {
+
+	protected static Log log = LogFactory.getLog(WebRendererReportsController.class);
+	
+	/**
+	 * Default Constructor
+	 */
+	public WebRendererReportsController() { }
+
+    @RequestMapping("/module/ugandaemrreports/reports/renderDataSet")
+    public void renderDataSet(ModelMap model, HttpServletRequest request, HttpServletResponse response,
+                              @RequestParam(required=false, value="uuid") String uuid,
+                              @RequestParam(required=false, value="name") String name,
+                              @RequestParam(required=false, value="type") Class<? extends DataSetDefinition> type) throws IOException {
+        try {
+            DataSetDefinitionService service = Context.getService(DataSetDefinitionService.class);
+            DataSetDefinition dataSetDefinition = service.getDefinition(uuid, type);
+            dataSetDefinition.setName(name);
+            dataSetDefinition.getParameters().clear();
+
+            ReportDefinition rd = new ReportDefinition();
+            rd.setName(name);
+            rd.getParameters().clear();
+
+
+
+            CsvReportRenderer renderer = new CsvReportRenderer();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
@dbaluku what the use of this ByteArrayOutPut Stream, seems to be redundant code